homeassistant/components/tesla/__init__.py

```diff
@@ -247,6 +249,7 @@ def device_info(self):
         """Prepare for unload."""
         pass
 
+    @Throttle(timedelta(seconds=MIN_UPDATE_INTERVAL))
```

Comment:
This throttle will only throttle this instance, not all the entities.

Please describe the update process in more detail and what is meant with queued updates.

Reply:
I think you're right this may be ineffective. The underlying API uses a lock on the update mechanism which also updates the time on the last successful update.  While repeated calls to the update function will exit immediately if it's called within a certain time of the successful update, a failure to update will immediately let the next update waiting in the queue to try to hit the API.

Reply:
Sounds like the update throttling should be handled completely in the library since there's where the lock, queue and failure mechanism is located.

Reply:
We can still modify the default scan interval for the polling entities if we think it's excessive. Then set `SCAN_INTERVAL` constant at the platform module level.

Reply:
Check out the latest recommendations on api data fetching:
https://developers.home-assistant.io/docs/en/integration_fetching_data.html

What category does the tesla integration fit into?

Reply:
You don't think doing it as a throttled class method works? SCAN_INTERVAL won't help because every entity is polled at that interval and each Tesla device has like 10 entities spawned and that will go up. Essentially, the current design in HA has forced the underlying API library to try to deal with that spammage. By adding the throttling in HA, it'd address that spamming.

This is a polling mechanism of the Tesla API and each of the entities polls the central controller built into the teslajsonpy API. I was trying a simple fix before redesigning the way the components work.

Reply:
It doesn't make sense to put the throttle here as long as the lock and logic around that is in the library.